pinot-segment-local/src/main/java/org/apache/pinot/segment/local/recordenricher/function/CustomFunctionEnricher.java

```diff
@@ -0,0 +1,57 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.pinot.segment.local.recordenricher.function;
+
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import org.apache.pinot.segment.local.function.FunctionEvaluator;
+import org.apache.pinot.segment.local.function.FunctionEvaluatorFactory;
+import org.apache.pinot.segment.local.recordenricher.RecordEnricher;
+import org.apache.pinot.spi.data.readers.GenericRow;
+
+
+public class CustomFunctionEnricher extends RecordEnricher {
+  private Map<String, FunctionEvaluator> _fieldToFunctionEvaluator;
+  private List<String> _fieldsToExtract;
+
+  @Override
+  public void init(Map<String, String> enricherProps) {
+    _fieldToFunctionEvaluator = new HashMap<>();
+    _fieldsToExtract = new ArrayList<>();
+    enricherProps.forEach((k, v) -> {
+      _fieldToFunctionEvaluator.put(k, FunctionEvaluatorFactory.getExpressionEvaluator(v));
+      _fieldsToExtract.addAll(_fieldToFunctionEvaluator.get(k).getArguments());
+    });
+  }
+
+  @Override
+  public List<String> getInputColumns() {
+    return _fieldsToExtract;
+  }
+
+  @Override
+  public void enrich(GenericRow record) {
+    _fieldToFunctionEvaluator.forEach((field, evaluator) -> {
+      record.putValue(field, evaluator.evaluate(record));
+    });
```

Comment:
Is is possible for an exception to get thrown by any of the `evaluator`s? If so, should we catch here (as was done in CLP) and log a message before propagating up?